src/fundus/scraping/crawler.py

```diff
@@ -212,8 +213,11 @@ def build_extraction_filter() -> Optional[ExtractionFilter]:
         for article in self._build_article_iterator(
             tuple(fitting_publishers), error_handling, build_extraction_filter(), url_filter
         ):
-            if not only_unique or article.html.responded_url not in response_cache:
-                response_cache.add(article.html.responded_url)
+            cached_url = article.html.responded_url
+            if any(parameter_indicator in cached_url for parameter_indicator in ("?", "#")):
+                cached_url = urljoin(article.html.responded_url, urlparse(article.html.responded_url).path)
```

Comment:
For readability, I would suggest making this a function `remove_query_parameter` and then rename `cached_url` in something alongside `url_without_query_parameter`